fem/bilininteg_vecdiffusion.cpp

```diff
@@ -506,33 +517,42 @@ void PAVectorDiffusionApply3D(const int NE,
 // PA Diffusion Apply kernel
 void VectorDiffusionIntegrator::AddMultPA(const Vector &x, Vector &y) const
 {
-   const int D1D = dofs1D;
-   const int Q1D = quad1D;
-   const Array<double> &B = maps->B;
-   const Array<double> &G = maps->G;
-   const Array<double> &Bt = maps->Bt;
-   const Array<double> &Gt = maps->Gt;
-   const Vector &D = pa_data;
-
-   if (dim == 2 && sdim == 3)
+#ifdef MFEM_USE_CEED
```

Comment:
There is a lot of places with `#ifdef MFEM_USE_CEED` followed by `if (DeviceCanUseCeed())`: couldn't we find a way (perhaps not in this PR) to avoid or generalize this 'difference' and `#ifdef` treatment?
Like `libCEED`, it would be good to be able to fuse the 3 kernels.